charts/everest/templates/check-crd.yaml

```diff
@@ -0,0 +1,22 @@
+# This template checks if all Everest CRDs are up-to-date during a Helm upgrade.
+# - It looks for CRDs with the label "everest.percona.com/crd".
+# - If no such CRDs are found, it fails the upgrade with an error message. Typically this may only ever haappen once, that is, when the chart is upgraded to the
+#   first version that includes these new labels on the CRDS.
+# - If the CRDs are found but their version (label "everest.percona.com/version") does not match the current chart version, it also fails the upgrade.
+# This ensures that CRDs are upgraded before the Helm release is upgraded.
+
+{{- if (and .Release.IsUpgrade .Values.upgrade.crdChecks )}}
+{{- $found := false }}
+{{- range $index, $crd := (lookup "apiextensions.k8s.io/v1" "CustomResourceDefinition" "" "" ).items }}
+  {{- if (and (hasKey $crd "metadata") (hasKey $crd.metadata "labels") (hasKey $crd.metadata.labels "everest.percona.com/crd")) }}
+    {{- $found = true }}
+    {{- $version := index $crd.metadata.labels "everest.percona.com/version" }}
+    {{- if ne $version $.Chart.Version }}
```

Comment:
Here we assume that if there is the "everest.percona.com/crd" then there should be the "everest.percona.com/version" as well. Should we also check the presenсe of the "everest.percona.com/version" label before trying to get the index? If the label does not exist, could it cause any problems when comparing the result with the Chart.Version in the template?

Reply:
Both labels are always set together, so I don't think there will ever be a case when the 'crd' label is there but not the 'version'.. Nevertheless, I added a check anyway since it is pretty straightforward to do